File: 88-MergeSortedArray/88-MergeSortedArray.c
// Last updated: 04/04/2026, 15:47:33
void merge(int* nums1, int nums1Size, int m,
           int* nums2, int nums2Size, int n) {

    int i = m - 1;          // last valid element in nums1
    int j = n - 1;          // last element in nums2
    int k = m + n - 1;      // last position in nums1

    while (i >= 0 && j >= 0) {
        if (nums1[i] > nums2[j]) {
            nums1[k--] = nums1[i--];
        } else {
            nums1[k--] = nums2[j--];
        }
    }

    while (j >= 0) {
        nums1[k--] = nums2[j--];
    }
}

File: 88-MergeSortedArray/88-MergeSortedArray.java
// Last updated: 04/04/2026, 15:47:28
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m-1;
        int j = n-1;
        int k = m+n-1;
        if(m==0) {
            for(int r = 0;r<nums1.length;r++){
                nums1[r]=nums2[r];  
            }
            return;
        }
        while (i>=0&&j>=0) {
            if(nums1[i]>nums2[j]){
                nums1[k]=nums1[i];
                k--;        
                i--;
            }
            else{
                nums1[k]=nums2[j];
                k--;
                j--;
            }
            }
        while(j>=0){
            nums1[k]=nums2[j];
            k--;
            j--;
        }
        }
}